cli/src/main/java/org/jboss/gm/cli/Main.java

```diff
@@ -111,8 +119,62 @@ public int run(String[] args) throws Exception {
         return result;
     }
 
-    private void executeGradle() throws ManipulationException {
+    private void logAndThrow(String message, Throwable cause) throws ManipulationException {
+        if (cause != null) {
+            logger.error(message, cause);
+            throw new ManipulationException(message, cause);
+        } else {
+            logger.error(message);
+            throw new ManipulationException(message);
+        }
+    }
+
+    private void verifyBuildEnvironment(ProjectConnection connection) throws ManipulationException {
+        try {
+            BuildEnvironment env = connection.getModel(BuildEnvironment.class);
+            String versionString = env.getGradle().getGradleVersion();
+            GradleVersion version = GradleVersion.version(versionString);
+
+            logger.info("Gradle version: {}", versionString);
+            logger.info("Java home: {}", env.getJava().getJavaHome());
+            logger.info("JVM arguments: {}", env.getJava().getJvmArguments());
+
+            if (version.compareTo(MIN_GRADLE_VERSION) < 0) {
+                String msg = String.format("%s is too old and is unsupported. You need at least %s.", version,
+                        MIN_GRADLE_VERSION);
+                logAndThrow(msg, null);
```

Comment:
So this could be
```
                throw new ManipulationException("{} is too old and is unsupported. You need at least {}.", version,
                        MIN_GRADLE_VERSION););
```

Reply:
I've changed all of the `throw new ManipulationException` to use this format.